internal/executor/exec.go

```diff
@@ -11,15 +11,19 @@ import (
 // The cmdOutput argument is used to print the command output.
 func Run(cmd []string, cmdOutput, cmdError bool, wg *sync.WaitGroup, outputCh, errorCh chan<- string) {
 	defer func() {
+		close(outputCh)
+		close(errorCh)
 		wg.Done()
 	}()
 
 	command := exec.Command(cmd[0], cmd[1:]...)
 	stdout, _ := command.StdoutPipe()
 	stderr, _ := command.StderrPipe()
 
-	//command.Wait()
-	command.Start()
+	if err := command.Start(); err != nil {
+		fmt.Printf("command execution error: %v\n", err)
+		return
```

Comment:
Why not sending the error to outputCh?

Other parts of the code does it

Reply:
I initially implemented it that way. After thinking a bit more it felt weird to report that error in the `outputCh` or `errorCh`. The reason behind this is that they will get wrapped by the caller, and printed with a prefix of `stdout:` or `stderr:` which doesn't make very clearly in telling us if we failed to spawn the command or if something failed inside the program we are running.

If you prefer I can proceed with using the output/err Ch. Let me know you opinion on that. In that case, should we ignore the `cmdOutput`/`cmdError` flags?

Reply:
I see your point @alessio-perugini, thanks for clarifying.
what do you think @ccoVeille, @89luca89 ?

Reply:
Why not, but then please consider:
- using fmt.Fprintf(os.Stderr to report the error
- check of the method signature shouldn't be updated to return an error. Handling error with printing is strange to me,it should be moved to higher level.
- check if other errors currently send to outputCh or errCh shouldn't also be refactored

Reply:
Agree. Since the function should be refactored we can just print it to stderr.

Reply:
I opened

- #63